src/rmarkdown/draft.ts

```diff
@@ -0,0 +1,134 @@
+import { QuickPickItem, QuickPickOptions, Uri, window, workspace } from 'vscode';
+import { extensionContext } from '../extension';
+import { executeRCommand, getCurrentWorkspaceFolder, getRpath, ToRStringLiteral, spawnAsync } from '../util';
+import * as cp from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as os from 'os';
+
+interface TemplateInfo {
+    id: string;
+    package: string;
+    name: string;
+    description: string;
+    create_dir: boolean;
+}
+
+interface TemplateItem extends QuickPickItem {
+    info: TemplateInfo;
+}
+
+async function getTemplateItems(cwd: string): Promise<TemplateItem[]> {
+    const lim = '---vsc---';
+    const rPath = await getRpath();
+    const options: cp.CommonOptions = {
+        cwd: cwd,
+        env: {
+            ...process.env,
+            VSCR_LIM: lim
+        }
+    };
+
+    const rScriptFile = extensionContext.asAbsolutePath('R/rmarkdown/templates.R');
+    const args = [
+        '--silent',
+        '--slave',
+        '--no-save',
+        '--no-restore',
+        '-f',
+        rScriptFile
+    ];
+
+    let items: TemplateItem[] = [];
+
+    try {
+        const result = await spawnAsync(rPath, args, options);
+        if (result.status !== 0) {
+            throw result.error || new Error(result.stderr);
+        }
+        const re = new RegExp(`${lim}(.*)${lim}`, 'ms');
+        const match = re.exec(result.stdout);
+        if (match.length !== 2) {
+            throw new Error('Could not parse R output.');
+        }
+        const json = match[1];
+        const templates = <TemplateInfo[]>JSON.parse(json) || [];
+        items = templates.map((x) => {
+            return {
+                alwaysShow: false,
+                description: `{${x.package}}`,
+                label: x.name,
+                detail: x.description,
+                picked: false,
+                info: x
+            };
+        });
+    } catch (e) {
+        console.log(e);
+        void window.showErrorMessage((<{ message: string }>e).message);
+    }
+
+    return items;
+}
+
+async function launchTemplatePicker(cwd: string): Promise<TemplateItem> {
+    const options: QuickPickOptions = {
+        matchOnDescription: true,
+        matchOnDetail: true,
+        canPickMany: false,
+        ignoreFocusOut: false,
+        placeHolder: '',
+        onDidSelectItem: undefined
+    };
+
+    const items = await getTemplateItems(cwd);
```

Comment:
Here we could indicate that a folder needs to be created by e.g.:
``` ts
    for(const item of items){
        if(item.info.create_dir){
            item.label = (item.label || '') + ' $(list-tree) ' ;
        }
    }
```

Reply:
Didn't know that icons could appear in the item label. `create_dir` is known in `getTemplateItems` when `label` is first specified. `$(new-folder)` might be more direct here.

Thanks for the tip!